### PR TITLE
Module2 cube

### PR DESCRIPTION
**Pull request by cnuebred**
finall version of checker palindrome

### CODE REVIEW

File: module2/homework/insensitivePalindrom/main.cpp [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#include "checkPalindrome.hpp"
+#include <iostream>
+
+int main(){
+    auto my_palindrome = is_palindrome("kaj..@@!#$@#$*ak1234    \nkaJaK");
+    std::cout << my_palindrome << std::endl;
+}
```
Comment (kyrtaq) [inline module2/homework/insensitivePalindrom/main.cpp]: new line!!!!!!!

File: module2/homework/insensitivePalindrom/checkPalindrome.hpp [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#pragma once
+#include <string>
+
+
+bool is_palindrome(std::string str);
```
Comment (kyrtaq) [inline module2/homework/insensitivePalindrom/checkPalindrome.hpp]: new line!!

**Review by kyrtaq — CHANGES_REQUESTED** [review]

File: module2/homework/insensitivePalindrom/CMakeLists.txt [hunk 3]
```diff
@@ -38,10 +38,10 @@ endif()
...
 
-add_executable(${PROJECT_NAME}-ut test.cpp) # add your cpp file here after test.cpp
+add_executable(${PROJECT_NAME}-ut test.cpp checkPalindrome.cpp) # add your cpp file here after test.cpp
 # if this is problematic take a look into CMakeLists.txt files in other exercises
 
 add_compile_options(${PROJECT_NAME}-ut -Wall -Wextra -Wconversion -pedantic -Werror)
 
 target_link_libraries(${PROJECT_NAME}-ut gtest_main)
-add_test(NAME ${PROJECT_NAME}_test COMMAND ${PROJECT_NAME}-ut)
+add_test(NAME ${PROJECT_NAME}_test COMMAND ${PROJECT_NAME}-ut)
```
Comment (NG90) [inline module2/homework/insensitivePalindrom/CMakeLists.txt]: new line!:P

File: module2/homework/insensitivePalindrom/checkPalindrome.cpp [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+                  return !std::isalpha(x) || std::isspace(x);
+              }),
+              str.end());
+
+    std::transform(str.begin(), str.end(), str.begin(), [](auto x) {
+        return std::tolower(x);
+    });
+
+    bool palindrome = std::equal(str.begin(), str.end(), str.rbegin());
+    return palindrome;
```
Comment (NG90) [inline module2/homework/insensitivePalindrom/checkPalindrome.cpp]: no need to create temporary variable `palindrome` just to return result. Please do it like this:
`return  std::equal(str.begin(), str.end(), str.rbegin());

File: module2/homework/insensitivePalindrom/checkPalindrome.cpp [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    str.erase(std::remove_if(str.begin(), str.end(), [](unsigned char x) {
+                  return !std::isalpha(x) || std::isspace(x);
+              }),
+              str.end());
+
+    std::transform(str.begin(), str.end(), str.begin(), [](auto x) {
+        return std::tolower(x);
+    });
+
+    bool palindrome = std::equal(str.begin(), str.end(), str.rbegin());
```
Comment (NG90) [inline module2/homework/insensitivePalindrom/checkPalindrome.cpp]: you can use const iterators here, like:
`std::equal(str.cbegin(), str.cend(), str.crbegin());`

Comment (NG90) [inline module2/homework/insensitivePalindrom/checkPalindrome.cpp]: It should also work if you check only half of the string:
` std::equal(str.cbegin(), str.cbegin() + str.size() / 2, str.crbegin());`

File: module2/homework/insensitivePalindrom/CMakeLists.txt [hunk 6]
```diff
@@ -38,10 +38,10 @@ endif()
 project(insensitivePalindrom)
 enable_testing()
 
-add_executable(${PROJECT_NAME}-ut test.cpp) # add your cpp file here after test.cpp
+add_executable(${PROJECT_NAME}-ut test.cpp checkPalindrome.cpp) # add your cpp file here after test.cpp
```
Comment (NG90) [inline module2/homework/insensitivePalindrom/CMakeLists.txt]: add also line that will compile main.cpp.
`add_executable(${PROJECT_NAME}  main.cpp checkPalindrome.cpp)`

**Review by NG90 — CHANGES_REQUESTED** [review]

**Review by NG90 — APPROVED** [review]

**Pull request merged**